File: backend/src/main/java/com/eventflow/modules/identity/api/dto/AuthDtos.java
package com.eventflow.modules.identity.api.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

/** DTOs espejo EXACTO de components.schemas del contrato (docs/api/05-openapi.yaml). Congelados. */
public final class AuthDtos {

    private AuthDtos() {
    }

    public record RegisterRequest(
            @NotBlank @Email @Size(max = 255) String email,
            @NotBlank @Size(min = 8, max = 72) String password,
            @NotBlank @Size(min = 1, max = 200) String fullName,
            @Pattern(regexp = "^\\+[1-9]\\d{6,14}$", message = "debe ser E.164") String phone) {
    }

    public record LoginRequest(@NotBlank @Email String email, @NotBlank String password) {
    }

    public record RefreshRequest(@NotBlank String refreshToken) {
    }

    public record LogoutRequest(@NotBlank String refreshToken) {
    }

    public record UserProfileDto(UUID id, String email, String fullName, String phone,
                                 List<String> roles, Instant createdAt) {
    }

    public record AuthTokensResponse(String accessToken, long accessTokenExpiresIn,
                                     String refreshToken, UserProfileDto user) {
    }
}

File: app/src/main/java/com/app/eventflow/data/remote/dto/AuthDtos.kt
package com.app.eventflow.data.remote.dto

import kotlinx.serialization.Serializable

/** DTOs espejo EXACTO de components.schemas (docs/api/05-openapi.yaml). Congelados. */

@Serializable
data class DataEnvelope<T>(val data: T)

@Serializable
data class RegisterRequestDto(
    val email: String,
    val password: String,
    val fullName: String,
    val phone: String? = null,
)

@Serializable
data class LoginRequestDto(val email: String, val password: String)

@Serializable
data class RefreshRequestDto(val refreshToken: String)

@Serializable
data class LogoutRequestDto(val refreshToken: String)

@Serializable
data class UserProfileDto(
    val id: String,
    val email: String,
    val fullName: String,
    val phone: String? = null,
    val roles: List<String> = emptyList(),
    val createdAt: String? = null,
)

/** Espejo del schema UpdateProfileRequest (solo campos editables: fullName requerido, phone E.164). */
@Serializable
data class UpdateProfileRequestDto(val fullName: String, val phone: String? = null)

@Serializable
data class AuthTokensResponseDto(
    val accessToken: String,
    val accessTokenExpiresIn: Long,
    val refreshToken: String,
    val user: UserProfileDto,
)
